### PR TITLE
Add copying helpers

### PR DESCRIPTION
**Pull request by ntrrg**
Since the sh.Copy helper just support one file operations, copying directories or multiple file may require help from the OS, which brings compatibility issues (one of the main reasons why we don't use Make). This PR aims to add the following helpers:

* `sh.CopyDir`: copies source directory content to the given destination. This will fail if destination doesn't exist.
* `sh.CopyDirAll`: same as above, but creates the destination directory if it doesn't exist.
* `sh.CopyAny`: copies the source elements to the given destination. Destination must be a directory.

Resolves: #211

### CODE REVIEW

**Comment by natefinch** [comment]
So, I was looking at this again, and I think copying the `cp` command is maybe not the best way to do this. I can't tell you how often I do the wrong thing with cp because I expect it to work one way and it works some other way.  The fact that the comment on the Cp function is as long as the function is a red flag, IMO.

I'd prefer multiple functions that were more obvious in how they work and explicitly fail if you use them incorrectly.

CopyContents - copies all the files and directories in an existing directory into another existing directory.
CopyDir - creates a new directory all files and directories in an existing directory into it (basically just mkdirall and then CopyContents)

And maybe CopyTo - takes a list of files and copies them to an existing directory.

**Comment by ntrrg** [comment]
Yes, actually it could be hard to follow the code some times :sweat_smile: I will try your approach

**Comment by ntrrg** [comment]
If you can, review the code again :) I will be working on the tests.

**Comment by ntrrg** [comment]
I think this may be implemented by users that really need it haha so I am closing the PR :sweat_smile:
